### DISCUSSION TITLE
BelongsToMany relationship queries don't have the "pivot" relation inside the model instance.

### DISCUSSION THREAD

**Discussion by OzanKurt** [yajra/laravel-datatables#2898] [General]
https://github.com/yajra/laravel-datatables/blob/8c909eae598fc487deeb252b912cf32b4d1bc803/src/EloquentDataTable.php#L35

Instead of that line we should probably update it to this:

```php
$this->query = $model instanceof Relation ? $model : $builder;
```

This way we can include the relation data in the result model instances.

**Comment by yajra** [comment]
https://github.com/yajra/laravel-datatables/blob/8c909eae598fc487deeb252b912cf32b4d1bc803/src/EloquentDataTable.php#L29-L30

Looks fine but maybe we should adjust the match case instead?

Also, what would be the output if changed to `$model`? The pivot values will be appended?
